### DISCUSSION TITLE
Menu Design Proposition for Agriculture Research Design and Data Analysis

### DISCUSSION THREAD

**Discussion by fran2or** [IDEMSInternational/R-Instat#6879] [Ideas]
With reference to #6620,  the special agriculture research menu in the current version in R-Instat is **Options by Context** but needs to change.  The new proposed name change is **Experimental** . The menu has not been fully conceptualized but the main proposed frame work  for its menu design is as follows:

**Experimental**
   - **Open From Library**
   - **Field Book Design**       
   - **Check Data**
      - One Variable summarise
      - One Variable Graph
      - One Variable Frequencies Tables
      - Duplicates
      - Compare Columns
      - Pivot Tables
   - **Prepare**
      - Column Operations
         - Row Summaries (Mean , Sum, Diference, etc)
         - Combine Columns
         - Recoding Columns
      - Calculate Difference Between Options
      - Stack
      - Unstack
      - Merge Additional Data
   - **Define** 
      - On - Station  
      - On  - Farm 
   - **Describe**
      - Frequency Tables
      - Summary Tables
      - Compare Two Options
      - Boxplot
      - Density Plot & Histograms 
      - Heat maps
   - **Model** 
     - Fit Model
     - Model Comparison

Attached is slightly  more detailed document. Fully prepared and approved dialogs will follow when ready.
[Experimental Data Design.docx](https://github.com/africanmathsinitiative/R-Instat/files/7399922/Experimental.Data.Design.docx)

**Comment by derekagorhom** [comment]
@fran2or from the first R script which is the Dataframe for Treatment;
the code runs well in R studio but when used in R-instat, we need to change some parts of the code so as to produce a dataframe in R-instat.
Below is the original rcode:

```
# Data Frame
# Multiple treatement
library("tidyr")
vec1 <- c("A", "B", "AA", "CDE")    # Create example vectors
vec2 <- 1:3
crossing(vec1, vec2) 

# Simple treatement
`data <- data.frame(data=matrix(data=NA, nrow=10, ncol=1))`
```

To make it run in R-instat this code should work 

```
# Create example vectors
vec1 <- c("A", "B", "AA", "CDE")    
vec2 <- 1:3

# Create a crossing of vec1 and vec2
crossed_data <- crossing(treat = vec1, yield = vec2)

# Import the data using data_book$import_data
data_book$import_data(data_tables = list(data3 = crossed_data))

# Remove the crossed_data object from the environment
rm(crossed_data)
```

But even in this case that is not how the R-instat code produces new data frames:
the example below shows how we create a dataframe in R-instat.

```
data <- data.frame(....Code....)
data_book$import_data(data_tables=list(data=data))
rm(data)
```

I will try and have a meeting with lily by Monday and discuss how we can make the rcode better.

**Reply by IssoufKon** [reply]
IssoufKon

**Reply by Emmanuel-Afrifa** [reply]
Well noted.

**Reply by lilyclements** [reply]
@derekagorhom which dialog do you see this in? Can't something like this do it:

```
vec1 <- c("A", "B", "AA", "CDE")
vec2 <- 1:3
crossed_data <- tidyr::crossing(treat = vec1, yield = vec2)
data_book$import_data(data_tables = list(data = crossed_data))
```

**Reply by derekagorhom** [reply]
@lilyclements this is not in any dialogue yet....we are trying to make an option in the create new dataframe dialogue which can perform this

**Reply by Emmanuel-Afrifa** [reply]
I just run the codes produced above in R-INSTAT and it produces a result.
![Screenshot (23)](https://github.com/user-attachments/assets/b2b543dd-bb4e-4495-8890-34a57c9309fc)

**Reply by lilyclements** [reply]
@derekagorhom thanks for explaining! So is it going to be as a new tab at the top of the "New Data Frame" dialog?

**Reply by derekagorhom** [reply]
@lilyclements yes please that is the plan since we currently do not have a way to create a dataframe for Agricultural research in R-instat

**Reply by lilyclements** [reply]
@derekagorhom Nice. Have you planned out the design of it?

**Reply by derekagorhom** [reply]
[Data for treatments.odt](https://github.com/user-attachments/files/18166551/Data.for.treatments.odt)
Yes @lilyclements above document contain the initail design. 
you can have a look

**Comment by IssoufKon** [comment]
**### Field book design**

```
library(agricolae)

**##Data : trt and trt1**
trt=c("CIP-101","CIP-201","CIP-301","CIP-401","CIP-501","CIP-601")
trt1=c("a","b","c","d","e")
r=3
kinds = "Super-Duper"
serie=2
#Bloc size
k=3
```

**Comment by IssoufKon** [comment]
**1. Completely randomized design**
```
#Use: design.crd(trt, r, serie = 2, seed = 0, kinds = "Super-Duper",randomization=TRUE)
outdesign <-design.crd(trt1,r,serie,2543,"Mersenne-Twister")
outdesign

# no seed
outdesign <-design.crd(trt,r,serie=3)
outdesign 
```
 
**2.Randomized complete block design**
```
#Use: design.rcbd(trt, r, serie = 2, seed = 0, kinds = "Super-Duper", first=TRUE,continue=FALSE,randomization=TRUE )
trt=c("F1V1","F1V2","F1V3","F2V1","F2V2","F2V3")
outdesign <-design.rcbd(trt,r=2,serie=2,986,"Wichmann-Hill") # seed = 986
outdesign
#book <-outdesign$book # field book
fieldbook <- zigzag(outdesign)
fieldbook
#print(outdesign$sketch)
#print(matrix(fieldbook[,1],byrow=TRUE,ncol=5))
```
   
**3. Latin square design**
```
#Use: design.graeco(trt1, trt2, serie = 2, seed = 0, kinds = "Super-Duper",randomization=TRUE)
outdesign <-design.lsd(trt,serie, seed=23)
outdesign
# Write on hard disk.
# write.table(lsd,"lsd.txt", row.names=FALSE, sep="\t")
# file.show("lsd.txt")
```
  
**4. Graeco-Latin designs**
```
#Use: design.graeco(trt1, trt2, serie = 2, seed = 0, kinds = "Super-Duper",randomization=TRUE)
outdesign <- design.graeco(trt,trt2,serie=1)
outdesign 
```
 
**5. Incomplete Latin Square Design or youden Design**
```
#Use: design.youden(trt, r, serie = 2, seed = 0, kinds = "Super-Duper",first=TRUE, randomization=TRUE)
outdesign <-design.youden(trt,r,serie,seed=23)
outdesign 
#youden <- outdesign$book
#print(outdesign$sketch)
#plots <-as.numeric(youden[,1])
#print(matrix(plots,byrow=TRUE,ncol=r))
#print(youden) # field book.
# Write on hard disk.
# write.table(youden,"youden.txt", row.names=FALSE, sep="\t")
# file.show("youden.txt")
```
  
 **6. Balanced Incomplete Block Designs**
```
# Use: design.bib(trt, k, r=NULL, serie = 2, seed = 0, kinds = "Super-Duper",maxRep=20,randomization=TRUE)
outdesign<-design.bib(trt,r=5, k, serie, seed =41, kinds="Super-Duper") # seed = 41
#print(outdesign$parameters)
#book<-outdesign$book
#plots <-as.numeric(book[,1])
#matrix(plots,byrow=TRUE,ncol=k)
#print(outdesign$sketch)
# write in hard disk
# write.csv(book,"book.csv", row.names=FALSE)
# file.show("book.csv")
```
    
**7. Cyclic designs**
```
#Use: design.cyclic(trt, k, r, serie = 2, rowcol = FALSE, seed = 0, kinds = "Super-Duper",randomization=TRUE
trt<-letters[1:8]
# block size = 2, replication = 6
outdesign1 <- design.cyclic(trt, k=3, r=3, serie)
outdesign1 
#names(outdesign1)
 # groups 1,2,3
#outdesign1$sketch[[1]]
#outdesign1$sketch[[2]]
#outdesign1$sketch[[3]]
#outdesign1$book
 # row-column design
#outdesign2<- design.cyclic(trt, k=2, r=6, serie=2, rowcol=TRUE)
#outdesign2$sketch
```
  
**8. Alpha designs**
```
# use : design.alpha(trt, k, r, serie = 2, seed = 0, kinds = "Super-Duper",randomization=TRUE)
trt<-1:30
outdesign<- design.alpha(trt,k,r,serie)  
outdesign
```
    
**9.	Augmented block designs**
```
#Use:design.dau(trt1, trt2, r, serie = 2, seed = 0, kinds = "Super-Duper", name="trt",randomization=TRUE)
trt<-c("A","B","C","D")
#trt1<-letters[20:26]
outdesign <-design.dau(trt,trt1, r, serie)
outdesign
```
   
**10. Split plot designs**
```
# Use: design.split(trt1, trt2,r=NULL, design=c("rcbd","crd","lsd"),serie = 2, seed = 0, kinds = "Super-Duper", first=TRUE,randomization=TRUE)
# 4 treatments and 5 blocks in split-plot
trt<-c("A","B","C","D") # have to be a character
trt1<-c(1,2,3) 
outdesign <-design.split(trt,trt1,r=3,serie=2,seed=45,kinds ="Super-Duper")#seed=45
outdesign 
# book<-outdesign$book# field book
# write in hard disk
# write.table(book,"book.txt", row.names=FALSE, sep="\t")
# file.show("book.txt")
```
    
**11. Split Plot Design**
```
#Use: design.split(trt1, trt2,r=NULL, design=c("rcbd","crd","lsd"),serie = 2,seed = 0, kinds = "Super-Duper", first=TRUE,randomization=TRUE)
outdesign <-design.split(trt,trt1, r=3,serie=2,seed=45,kinds ="Super-Duper")#seed=45
outdesign
book<-outdesign$book# field book
# write in hard disk
# write.table(book,"book.txt", row.names=FALSE, sep="\t")
# file.show("book.txt")
design.strip 
```
   
**12. Strip Plot Design**
```
#Use:design.strip(trt1, trt2,r, serie = 2, seed = 0, kinds = "Super-Duper",randomization=TRUE)
trt=c("A","B","C","D","E","F")
outdesign <-design.strip(trt,trt1,r, serie=2,seed=45,kinds ="Super-Duper") # seed = 45
#book <-outdesign$book # field book
# write in hard disk
# write.table(book,"book.txt", row.names=FALSE, sep="\t")
# file.show("book.txt")
```
  
**13. Design of experiments for a factorial**
```
#Use: design.ab(trt, r, serie = 2, design=c("rcbd","crd","lsd"), seed = 0, kinds = "Super-Duper",first=TRUE,randomization=TRUE)
# factorial 3 x 2 with 3 blocks
trt<-c(3,2) # factorial 3x2
outdesign <-design.ab(trt, r=3, serie=2)
outdesign 
book<-outdesign$book
head(book,10) # print of the field book
```

**Reply by lilyclements** [reply]
This looks really interesting @IssoufKon. Are these suggestions for a dialog which will generate the data for this?

How do you propose we offer these options? I can't remember the differences between, say, a completely randomised design and a randomised complete block design, but, perhaps it would be good for the user to say what _they_ want, and for us to offer the appropriate design as a result (so it's guided by analysis not by definition). Similar to how our modelling dialogs would work - the user would put in their variables, and we offer the appropriate model for it. Or similar to how we do our One Var Graph/Summarise where the appropriate graphs and statistics are offered.

On some tricot analyses we are doing, I've come across a function `randomize` which is an "incomplete block A-optional design". I'm not too sure on how this differs from an incomplete block design, etc, but thought I'd run it by you here in case either you knew, and/or if it were already covered in your 13 options.

**Reply by IssoufKon** [reply]
Hi @lilyclements,
We have used the design types in the agricolae package. The difference between “Completely Randomized Design” and “Randomized Complete Block Design” is that “Completely Randomized Design” generates a completely randomized design with equal or different repetitions, whereas “Randomized Complete Block Design” generates a completely randomized block design. We'd like the user to fill in the various boxes, and based on this, only those design types corresponding to these criteria will appear. The dialog is intended to produce design data that can then be used for graphic visualization.

**Reply by lilyclements** [reply]
Great, so how can we  put that on a dialog as a different option? Perhaps if they fill out the "Block" receiver it gives a RCBD and otherwise gives a CRD? 

Let's continue on this! How does a Latin Square differ to a Greco-Latin Square?

**Reply by IssoufKon** [reply]
Hi @lilyclements ,
I share your idea on this point "how can we put that on a dialog as a different option? Perhaps if they fill out the "Block" receiver it gives a RCBD and otherwise gives a CRD?". 
The difference between Greco-Latin Square and Latin Square is that : A Latin square design is a grid of letters where each letter appears once in each row and column, while a Graeco-Latin square design is created by combining two Latin squares. I'm still wondering how to materialize this difference on r-instat.

**Reply by lilyclements** [reply]
@IssoufKon nice! Thanks for this. This is great, and our discussion in the meeting earlier was really useful. 

So from what I can see:

| Design  | Description | Function |
| ------------- | ------------- | ------------- |
| Latin | You have one factor, and everything is used exactly once on each row and column (Full and Balanced)  | `design.lsd` |
| Graeco-Latin | You need to control for two different factors at the same time. So we can say something about the factors to differentiate this (Paired?) | `design.graeco` |
| Incomplete Latin Square | Not all factors are used on each row/column (Incomplete and Balanced) | `design.youden` |

All three of `design.lsd`, `design.graeco`, and `design.youden` functions need to be given a `trt`, `serie`, (and `seed`). 

Then `design.youden` needs `r`, which is your number of columns (or replicates, depending how we want to see it!). **If `r` is equal to your number of treatments, then you just have a latin square design!** So to differentiate these two, we can just simply ask the number of replicates. If the number of replicates (r) equals the number of factors in `trt`, then we run `design.lsd`, else we run `design.graeco`.

So, one idea of how we could do this is to have a checkbox for "Paired Design".
If unchecked, they fill out the Treatment (`trt`) argument
They can fill out the Replicates (`r`) with a nud, default as the number of factors given in the `trt`. This nud for `r` cannot be greater than the number of factors, and cannot be less than 2. 
If `r` = number of factors in `trt`, then run `design.lsd`
If `r` < number of factors in `trt`,  then run `design.youden`

If the checkbox for "Paired Design" is checked, they fill out the Treatment (`trt1`) argument
And a new option comes for `trt2`, to fill out the second Treatment argument
(and the Replicates option is no longer visible)
This then runs `design.graeco`.

Does this make sense?

We can start looking at these different block designs and how to differentiate them - but only if what we've done so far makes sense

**Comment by derekagorhom** [comment]
https://github.com/IDEMSInternational/R-Instat/issues/9459
Hello guys the above link goes to the first feature for agricultural research design (the plot option dialogue)

**Comment by rdstern** [comment]
@lilyclements and @IssoufKon I add some related comments in the interst of some quick progress.

a) On the design dialog we don't need to include all the options that are in agricolae.  The dialog needs to help those who would like to design and picture their field plan for some of the common and useful designs.  I would not mind if graeco-latin squares - even maybe Latin squares - we not included initially.  We need randomised blocks and split plots, and maybe even split-split plots - that I don't like them.
We need to allow factorial treatment structure and NOT assume that multiple factors imply split plot designs.  Useful to have some incomplete block designs.
I really like the example dataset in the agridat package called `ryder.groundnut`.  This shows the value of including variables so the field plan can be shown when appropriate and the code supplied shows the value of the desplot package.  Might you be including it?
The code supplied in agridat for the analysis of this dataset is interesting, partly because it shows that a possible error in the coding of the data may have greatly influenced the results.  It is also a nice example for teaching which I hope we can use in a forthcoming workshop.
